### ISSUE TITLE
Unable to use add_url_rule() directly

### ISSUE THREAD

**Issue by AbdealiLoKo** [marshmallow-code/flask-smorest#269]
Currently, in one of my projects we use `blueprint.add_url_rule()` and we are trying to move towards using smorest.

When we tried this - it seems like the documentation is not getting updated when we use `add_url_rule`.
On digging further, I see that `blueprint.route()` is doing some extra logic which is not happening if I directly try to add the rule

https://github.com/marshmallow-code/flask-smorest/blob/master/flask_smorest/blueprint.py#L127

This is a blocker for us - as we have our own custom classes that we are using - kind of like the Django viewsets.
So, basically we cannot decorate the class.
For example: We have a class `MyClass` and we create 2 views from it:
 - `MyClass.create_view(action={'get': 'list' , 'post': create'})` for `/`
 - and `MyClass.create_view(action={'post': 'edit', 'delete': 'delete'})` for `/<id>`

It would be good to have a function to do this logic instead of just a decorator

**Comment by lafrech** [comment]
I suppose you could use `route` the non-decorating way, couldn't you (not sure, confirmation welcome)?

```py
    blp.route(...)(view_func)
```

Anyway, I just sent a PR to allow the use of `add_url_rule`. @AbdealiJK do you think you could  try it?

**Comment by AbdealiLoKo** [comment]
@lafrech apologies for the delayed reply but thanks a lot for looking into it !
I'll test this out tomorrow and get back

**Comment by AbdealiLoKo** [comment]
Works for me - I've added a minor comment in the PR
The route() function is now exactly the same as the function provided by Flask's Scaffold class. So, it does not need to be overridden anymore.
